src/components/Layout/Topbar/TopBarLink.jsx

```diff
@@ -0,0 +1,28 @@
+import { Button, Text } from '@chakra-ui/react';
+import { Link } from 'react-router-dom';
+
+function TopBarLink({ handleLogout, text, index }) {
+  return (
+    <Link
+      key={index}
+      to={`/${text}`}
+      onClick={() => {
+        if (text === 'logout') handleLogout();
```

Comment:
This check here is not a good solution - imagine if we had 10 links, all with different behaviour, we would have to add 10 ifs here to check which behaviour we want - instead, a better solution would be to define all the links with their behaviour in the parent component, so instead of:

```jsx
const guestLinks = ['login', 'register'];
const userLinks = ['logout', 'my-profile'];
```

you'd have

```jsx
const guestLinks = [{
    text: 'Login',
    href: '/login',
}, {
    text: 'Logout',
    onClick: () => handleLogout(),
}];
```

and in this component:

```jsx
<Link to={href ?? '#'} onClick={onClick ?? () => {}} />
```

P.S. `key` is not required here, it's only required in direct children of `map`, which you have in the parent component already :)